packages/devtools_app/lib/src/shared/table.dart

```diff
@@ -845,34 +842,32 @@ class _TableState<T> extends State<_Table<T>> with AutoDisposeMixin {
     sortColumn = widget.sortColumn;
     sortDirection = widget.sortDirection;
     scrollController = ScrollController();
-
+    _addScrollListener(widget.selectionNotifier);
     _initSearchListener();
   }
 
-  @override
-  void didUpdateWidget(_Table<T> oldWidget) {
-    super.didUpdateWidget(oldWidget);
+  void _addScrollListener(ValueNotifier<Selection<T>>? selectionNotifier) {
+    if (selectionNotifier != null) {
+      addAutoDisposeListener(selectionNotifier, () {
+        WidgetsBinding.instance.addPostFrameCallback((_) {
```

Comment:
instead of putting this entire thing in a post frame callback, I would add a parameter `scrollInPostFrameCallback` to `scrollToPosition`, and when true, wrap only the call to `scrollController.animateTo(...)` in a post frame callback. 

Otherwise, all these calculations will be delayed until a frame has been drawn, and we don't need that - we just want the scroll to take place after a frame has been drawn

Reply:
I believe we need the calculation to be delayed since we need to calculate the nodeIndex after the parents have been expanded.